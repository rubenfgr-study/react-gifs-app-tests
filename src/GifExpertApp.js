import React, { Fragment, useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  //   const [categories, setCategories] = useState(["Goku"]);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <br />
      <AddCategory setCategories={setCategories} />
      <br />
      {categories.map((value, index) => (
        <GifGrid key={value} category={value} />
      ))}
    </Fragment>
  );
};

export default GifExpertApp;
