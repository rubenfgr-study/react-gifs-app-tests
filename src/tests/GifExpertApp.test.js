import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("GifExpertApp Test", () => {
  test("GifExpertApp test snapshoot", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar una lista de categories", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
