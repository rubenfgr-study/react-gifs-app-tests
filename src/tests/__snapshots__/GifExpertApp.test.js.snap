// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Test GifExpertApp test snapshoot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
</Fragment>
`;

exports[`GifExpertApp Test debe mostrar una lista de categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;
